middleware/multipart.js

```diff
@@ -91,10 +95,37 @@ module.exports = function (config) {
                 return;
             }
 
-            req.body = fields;
-            req.files = files;
-            res.once('finish', cleanify(files));
+            // Use firstValues to handle formidable 3.x array format
+            req.body = firstValues(form, fields);
```

Comment:
Is the `firstValues` a potentially expensive operation? Should we considering storing a reference to reuse on L99-L100?

Reply:
99 & 100 are different calls. I would say it's likely the perf difference would be minimal anyways.